ibis/backends/duckdb/__init__.py

```diff
@@ -175,8 +176,19 @@ def fetch_from_cursor(
         cursor: duckdb.DuckDBPyConnection,
         schema: sch.Schema,
     ):
+        import pyarrow.types as pat
+
         table = cursor.cursor.fetch_arrow_table()
-        df = table.to_pandas(timestamp_as_object=True)
+        df = pd.DataFrame(
+            {
+                name: (
+                    col.to_pylist()
```

Comment:
I *think* this properly handles `timestamp_as_object=True`, but it'd be good to check this.

Reply:
Looks like it works as expected:

```
In [16]: import datetime, pandas as pd, pyarrow as pa

In [17]: t = pa.Table.from_pandas(pd.DataFrame({'a': [datetime.datetime(5600, 12, 1)]}))  # pd.Timestamp fails with out of bounds

In [18]: t
Out[18]:
pyarrow.Table
a: timestamp[us]
----
a: [[5600-12-01 00:00:00.000000]]

In [19]: t['a'].to_pylist()
Out[19]: [datetime.datetime(5600, 12, 1, 0, 0)]
```

Reply:
What was the issue here?

Reply:
`rich` won't handle truncation and indentation of numpy arrays, so any nested data needs to be a non-ndarray Python object.